components/dashboard/filters/OrderFilter.tsx

```diff
@@ -1,73 +1,76 @@
 import React from 'react';
 import { ArrowDownNarrowWide, ArrowDownWideNarrow } from 'lucide-react';
-import { defineMessage, useIntl } from 'react-intl';
+import { defineMessage, MessageDescriptor, useIntl } from 'react-intl';
 import { z } from 'zod';
 
-import { FilterConfig } from '../../../lib/filters/filter-types';
+import { FilterComponentProps, FilterConfig } from '../../../lib/filters/filter-types';
 
 import { Button } from '../../ui/Button';
 import { Select, SelectContent, SelectItem, SelectTrigger } from '../../ui/Select';
 
-const OrderFilterSchema = z.enum(['CREATED_AT,DESC', 'CREATED_AT,ASC']).default('CREATED_AT,DESC');
-
-type OrderFilterValue = z.infer<typeof OrderFilterSchema>;
-const i18nOrderByLabel = (intl, value) => {
-  const labels = {
-    'CREATED_AT,DESC': intl.formatMessage({ id: 'ExpensesOrder.NewestFirst', defaultMessage: 'Newest First' }),
-    'CREATED_AT,ASC': intl.formatMessage({ id: 'ExpensesOrder.OldestFirst', defaultMessage: 'Oldest First' }),
-  };
-  return labels[value];
-};
-
 const parseChronologicalOrderInput = str => {
   const [field, direction] = str.split(',');
   return { field, direction };
 };
 
-function toApiVariables(value, key) {
-  return { [key]: parseChronologicalOrderInput(value) };
-}
-
-export const orderByFilter: FilterConfig<z.infer<typeof OrderFilterSchema>> = {
-  schema: OrderFilterSchema,
-  toVariables: toApiVariables,
-  filter: {
-    labelMsg: defineMessage({ id: 'OrderBy', defaultMessage: 'Order by' }),
-    static: true,
-    StandaloneComponent: OrderFilter,
-    valueRenderer: ({ value, intl }) => i18nOrderByLabel(intl, value),
+export const orderByFilter = buildOrderByFilter(
```

Comment:
@gustavlrsn I built an abstraction to the order filter here, as I wanted to use it to order on something other than created at.